faculty/clients/experiment.py

```diff
@@ -83,7 +83,8 @@ class ExperimentRunStatus(Enum):
     ],
 )
 
-Metric = namedtuple("Metric", ["key", "value", "timestamp"])
+Metric = namedtuple("Metric", ["key", "value", "timestamp", "step"])
+Metric.__new__.__defaults__ = (None,)
```

Comment:
It took me a while to realize that this sets the default value for the *last* field of the namedtuple, as opposed to the first one. I'm still not sure why it does that, and the pattern of using `__new__.__defaults__` is not mentioned in the docs, so I would refrain from using it.

A better (IMO) pattern would be the one given in [this stackoverflow answer](https://stackoverflow.com/a/16721002) because it exposes that `step` is the optional attribute with a default value of `None` in the `Metric` class:

```python
class Metric(namedtuple("Metric", ["key", "value", "timestamp", "step"])):
    __slots__ = ()
    def __new__(cls, key, value, timestamp, step=None):
        return super(Node, cls).__new__(cls, key, value, timestamp, step)
```

Reply:
Also, it's not obvious to me that having a default is necessary at all. The marshmallow schema below will already fill in `None` if the response from the service does not have a step, so the actual namedtuple can require having `step` set.

Reply:
We have that default in `Metric` because of its use in mlflow-faculty. At the moment mlflow (0.9) and mlflow-faculty don't have the concept of step. This default allows it to work with auto-increasing steps in the backend set by atlas.

When we upgrade to mlflow 1.0, we should also update mlflow-faculty. At that point in mlflow they send step=0  by default (incidentally the reason for that error in the docs string)